File: src/main/java/seremis/geninfusion/world/GIWorldGenerator.java
package seremis.geninfusion.world;

import cpw.mods.fml.common.IWorldGenerator;
import net.minecraft.world.World;
import net.minecraft.world.chunk.IChunkProvider;
import net.minecraft.world.gen.feature.WorldGenMinable;
import seremis.geninfusion.block.ModBlocks;

import java.util.Random;

public class GIWorldGenerator implements IWorldGenerator {

    @Override
    public void generate(Random random, int chunkX, int chunkZ, World world, IChunkProvider chunkGenerator, IChunkProvider chunkProvider) {
        switch(world.provider.dimensionId) {
            case -1:
                generateNether(world, random, chunkX * 16, chunkZ * 16);
            case 0:
                generateSurface(world, random, chunkX * 8, chunkZ * 8);
            case 1:
                generateEnd(world, random, chunkX * 16, chunkZ * 16);
        }
    }

    private void generateSurface(World world, Random random, int blockX, int blockZ) {
        for(int i = 0; i < 7; i++) {
            int xCoord = blockX + random.nextInt(1);
            int yCoord = random.nextInt(60);
            int zCoord = blockZ + random.nextInt(1);
            new WorldGenMinable(ModBlocks.oreTitanium, 10).generate(world, random, xCoord, yCoord, zCoord);
        }
    }

    private void generateNether(World world, Random random, int blockX, int blockZ) {

    }

    private void generateEnd(World world, Random random, int blockX, int blockZ) {

    }
}

File: src/main/scala/seremis/geninfusion/world/GIWorldGenerator.scala
package seremis.geninfusion.world

import java.util.Random

import cpw.mods.fml.common.IWorldGenerator
import net.minecraft.world.World
import net.minecraft.world.chunk.IChunkProvider
import seremis.geninfusion.world.generate.WorldGenCrystal

class GIWorldGenerator extends IWorldGenerator {

    val crystalGen = new WorldGenCrystal()

    override def generate(random: Random, chunkX: Int, chunkZ: Int, world: World, chunkGenerator: IChunkProvider, chunkProvider: IChunkProvider) {
        world.provider.dimensionId match {
            case -1 => generateNether(world, random, chunkX * 16, chunkZ * 16)
            case 1 => generateEnd(world, random, chunkX * 16, chunkZ * 16)
            case default => generateSurface(world, random, chunkX * 8, chunkZ * 8)
        }
    }

    private def generateSurface(world: World, random: Random, blockX: Int, blockZ: Int) {
        for(i <- 0 until 1) {
            val x = blockX + random.nextInt(16)
            val z = blockZ + random.nextInt(16)

            crystalGen.generate(world, random, x, 10, z)
        }
    }

    private def generateNether(world: World, random: Random, blockX: Int, blockZ: Int) {}

    private def generateEnd(world: World, random: Random, blockX: Int, blockZ: Int) {}
}